salt/returners/slack_returner.py

```diff
@@ -206,10 +209,17 @@ def returner(ret):
         log.error('slack.api_key not defined in salt config')
         return
 
+    if only_show_failed and changes:
+        log.error('cannot define both slack.changes and slack.only_show_failed in salt config')
+        return
+
     returns = ret.get('return')
     if changes is True:
         returns = dict((key, value) for key, value in returns.items() if value['result'] is not True or value['changes'])
 
+    if only_show_failed is True:
+        returns = dict((key, value) for key, value in returns.items() if value['result'] is not True)
```

Comment:
This should be switched to dictionary interpretation, since we no longer support py2.6.

`{key, value for key, value in returns.items() if value['result'] is not True}`

This is significantly faster.

Reply:
@gtmanfred Good to know, updated lines 218 and 221 with new method.